include/pushmi/o/extension_operators.h

```diff
@@ -48,29 +48,27 @@ constexpr decltype(auto) apply(F&& f, Tuple&& t) {
 
 namespace detail {
 
-template <class... TagN>
+template <class Cardinality, bool IsFlow = false>
```

Comment:
is it better to use a bool than a type. There will also be a param added for time eventually..

Reply:
perhaps param should be all the properties and the specialization should be selected by requires expressions?

Reply:
Perhaps the intent of class... TagN should have been commented :) I intended for there to be an argument per property category. So Cardinality, Flow and Time are categories..

Reply:
I considered that, but the properties are unordered, and I didn't want duplicate template instantiations for `parameter<B,A>` and `properties<B,A>`.

As for whether a `bool` is appropriate here, I didn't want to force the use of `std::conditional<propert_query_v<....>, X, void>`. This seems cleaner.

Reply:
yes, it is cleaner at the callsite. I just avoid using true/false as a parameter at runtime or compiletime because you have to refer to the declaration of the parameter to know what true and false mean..

Reply:
I do too, but this stuff is local. Only `out_from_fn` is used anywhere else, and no nastiness is exposed.